lib/active_admin/async_count.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+module ActiveAdmin
+  class AsyncCount
+    def initialize(collection)
+      @collection = collection.except(:select, :order)
+      @promise = @collection.async_count
+    end
+
+    def count
+      value = @promise.value
+      # value.value due to Rails bug https://github.com/rails/rails/issues/50776
+      value.respond_to?(:value) ? value.value : value
```

Comment:
I'm not entirely sure what a "good" solution is to this.  Rails 7.1.3 has a bug with its query cache such that promises get wrapped by an "extra" promise.  It doesn't seem like ActiveAdmin's responsibility to work around that, but I'm not sure what alternatives there might be.  In any case, this is isolated to this class and is tested, so that seemed like a good starting point.

Reply:
Implementing the workaround here makes sense. We don't know the Rails version the host app is running.